src/Castle.Core/DynamicProxy/Contributors/MembersCollector.cs

```diff
@@ -178,10 +178,7 @@ protected bool AcceptMethod(MethodInfo method, bool onlyVirtuals, IProxyGenerati
 		/// </remarks>
 		protected bool AcceptMethodPreScreen(MethodInfo method, bool onlyVirtuals, IProxyGenerationHook hook)
 		{
-			if (IsInternalAndNotVisibleToDynamicProxy(method))
-			{
-				return false;
-			}
+			// NOTE: at this point, the method's accessibility should already have been checked (see `AddMethod` above)
```

Comment:
The only code execution paths leading here look like this:

`AddMethod` &rarr; the various collectors' `GetMethodToGenerate` methods ( &rarr; `AcceptMethod` ) &rarr; `AcceptMethodPreScreen`